src/node/node_state.h

```
@@ -1653,6 +1661,12 @@ namespace ccf
         ShareManager::clear_submitted_recovery_shares(tx);
         service_info->status = ServiceStatus::WAITING_FOR_RECOVERY_SHARES;
         service->put(service_info.value());
+        auto unsealed_ls = try_unseal_ledger_secret();
```

I think this is too lax and we should be stricter. If the configuration has a `sealed_ledger_secret`, then we should try to use that and fail loudly if that's not possible (eg - abandon this recovery if the crypto fails).

---

I've upgraded everything to throw on failed unsealing etc.